### PR TITLE
PR: Ovito visualisation

### PR DESCRIPTION
**Pull request by sblackburn86**
Minimal script to convert the output of an MTP evaluation to a 'movie' in OVITO. A README explains how to color code based on the MaxVol gamma values. 

We can introduce additional filters if we want to isolate specific atoms / clusters. This is something that could be done in a separate (future) PR.

### CODE REVIEW

File: data/mtp_templates/mtp_predictions.csv
Comment (rousseab) [inline data/mtp_templates/mtp_predictions.csv:1]: Same comment as for `dump.si-10000-4.yaml`.

File: examples/local/mtp_to_ovito.sh [hunk 1]
```diff
@@ -0,0 +1,10 @@
+#!/bin/bash
+
+MTP_PREDICTION=./experiments/mtp_find_region/predictions.csv
+LAMMPS_OUTPUT=./experiments/mtp_find_region/dump.si-10000-4.yaml
```
Comment (rousseab) [inline examples/local/mtp_to_ovito.sh:4]: This filename is identical to a file now located in `data/mtp_template/mtp_predictions.csv`. Why not move that file here?

File: examples/local/mtp_to_ovito.sh [hunk 2]
```diff
@@ -0,0 +1,10 @@
+#!/bin/bash
+
+MTP_PREDICTION=./experiments/mtp_find_region/predictions.csv
```
Comment (rousseab) [inline examples/local/mtp_to_ovito.sh:3]: This file doesn't exist. Should it be `data/mtp_templates/mtp_predictions.csv`? Why not move that file here?

**Review by rousseab — CHANGES_REQUESTED** [review]
The ovito viz looks pretty cool!

I make a few suggestions to improve the usability of the ovito data generation script.

File: examples/local/mtp_to_ovito.sh [hunk 3]
```diff
@@ -0,0 +1,10 @@
...
+LAMMPS_OUTPUT=./experiments/mtp_find_region/dump.si-10000-4.yaml
+OVITO_OUTPUT=./test_si_structure_ovito.xyz
+
+python crystal_diffusion/analysis/ovito_visualisation.py \
```
[the remark this answers is not in the document]
Comment (sblackburn86) [reply examples/local/mtp_to_ovito.sh:7]: I'll add a train_mtp entry point with a .sh script to generate those

File: data/mtp_templates/dump.si-10000-4.yaml
[the remark this answers is not in the document]
Comment (sblackburn86) [reply data/mtp_templates/dump.si-10000-4.yaml:1]: moved to examples

File: examples/local/mtp_to_ovito.sh (hunk 1 above)
Comment (sblackburn86) [reply examples/local/mtp_to_ovito.sh:4]: moved

File: examples/local/mtp_to_ovito.sh (hunk 2 above)
Comment (sblackburn86) [reply examples/local/mtp_to_ovito.sh:3]: yes, moved to examples/

File: crystal_diffusion/analysis/ovito_visualisation.py [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+import numpy as np
+import pandas as pd
+import yaml
+
+
+def main():
+    """Read MTP output files and convert to xyz format readable by OVITO."""
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--prediction_file", help="MTP prediction files. Should contain position and MaxVol gamma")
+    parser.add_argument("--lammps_output", help="MTP prediction files. Should contain position and MaxVol gamma")
```
[the remark this answers is not in the document]
Comment (sblackburn86) [reply crystal_diffusion/analysis/ovito_visualisation.py:18]: good catch, fixed

File: crystal_diffusion/analysis/ovito_visualisation.py [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+
+    Returns:
+        lattice: 3x3 array with lattice coordinates
+    """
+    with (open(lammps_output_file, 'r') as f):
+        l_yaml = yaml.safe_load_all(f)
+        for d in l_yaml:  # loop over LAMMPS outputs to get the MD box - we only need the first step
+            # lattice in yaml is 3 x 2 [0, x_lim]
+            # we assume a rectangular lattice for now with the 2nd coordinates as the lattice vectors
+            print(d)
```
[the remark this answers is not in the document]
Comment (sblackburn86) [reply crystal_diffusion/analysis/ovito_visualisation.py:45]: debug thing I forgot to rm

File: tests/analysis/test_ovito_visualisation.py [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+
+import numpy as np
+import pandas as pd
+import pytest
+
+from crystal_diffusion.analysis.ovito_visualisation import (
+    get_lattice_from_lammps, mtp_predictions_to_ovito)
+
+
+@pytest.fixture
```
[the remark this answers is not in the document]
Comment (sblackburn86) [reply tests/analysis/test_ovito_visualisation.py:12]: leftover from rewriting the test in a class

File: tests/analysis/test_ovito_visualisation.py [hunk 7]
```diff
@@ -0,0 +1,106 @@
...
+box:
+    - [0, 4.0]
+    - [0, 5.0]
+    - [0, 6.0]
+        """
+        with open(os.path.join(tmpdir, "test_output.yaml"), "w") as f:
+            f.write(lattice_output)
+        return os.path.join(tmpdir, "test_output.yaml")
+
+    def test_get_lattice_from_lammps(self, lammps_output, tmpdir):
```
[the remark this answers is not in the document]
Comment (sblackburn86) [reply tests/analysis/test_ovito_visualisation.py:50]: you're right. Removed

**Review by rousseab — APPROVED** [review]
Good to go :+1:

**Pull request merged**